\documentclass[conference]{IEEEtran}
\IEEEoverridecommandlockouts
% The preceding line is only needed to identify funding in the first footnote. If that is unneeded, please comment it out.

\usepackage{amsmath}
\usepackage{todonotes}
\usepackage{algorithm}
\usepackage[noend]{algpseudocode}
\usepackage{cite}
\usepackage{amsmath,amssymb,amsfonts}
\usepackage{algorithmic}
\usepackage{graphicx}
\usepackage{textcomp}
\usepackage{xcolor}
\def\BibTeX{{\rm B\kern-.05em{\sc i\kern-.025em b}\kern-.08em
    T\kern-.1667em\lower.7ex\hbox{E}\kern-.125emX}}
    
\usepackage{atbegshi}% http://ctan.org/pkg/atbegshi
\AtBeginDocument{\AtBeginShipoutNext{\AtBeginShipoutDiscard}}


\begin{document}

\title{ Application Note: $\mu$Polar \-- An Interactive 2D Visualization Tool for  Microfluidics and  Microscopic Time Series Images }

% \author{\IEEEauthorblockN{1\textsuperscript{st} Mehran Ghafari}
% \IEEEauthorblockA{\textit{SimCenter, Dept. Computer Science and Engineering} \\
% \textit{University of Tennessee, Chattanooga, USA}\\
% ryg668@mocs.utc.edu}
% \and
% \IEEEauthorblockN{2\textsuperscript{nd} Hao-Bo Guo}
% \IEEEauthorblockA{\textit{SimCenter, Dept. Computer Science and Engineering} \\
% \textit{University of Tennessee, Chattanooga, USA}\\
% haobo-guo03@utc.edu}
% \and
% \IEEEauthorblockN{3\textsuperscript{rd} Weiwei Dang}
% \IEEEauthorblockA{\textit{Dept. Molecular and Human Genetics, Huffington Center on Aging}\\
% \textit{Baylor College of Medicine, Houston, USA}\\
% Weiwei.Dang@bcm.edu}
% \and
% \IEEEauthorblockN{4\textsuperscript{th} Hong Qin}
% \IEEEauthorblockA{\textit{SimCenter, Dept. Computer Science and Engineering}\\
% {Dept. Biology, Geology and Enviromental Science} \\
% \textit{University of Tennessee, Chattanooga, USA}\\
% hong-qin@utc.edu}
% }

\maketitle


\begin{Summary}

Microfluidics-based microscopy is becoming an increasingly popular research tool to monitor cellular events in biomedical research. One such application is the high-throughput analysis of cells division. It is challenging to visualize and interpret the data gathered through microfluidics-based microscopy. Here, we developed a circular plotting method, $\mu$Polar, to visualize the trends and patterns of the cell movements and cellular division events at different time-points. Our method is interactive and easy to use. We demonstrated the utility of our method by visualizing the events of dividing yeast cells and migrating mouse fibroblasts. Our method could be potentially applied to other types of microfluidic devices including microscopic images. This software is implemented in an R package $\mu$Polar\ available through GitHub.


\end{Summary}


\begin{IEEEkeywords}
Microfluidics devices, microscopic data, cell distance, $\mu$Polar visualization 
\end{IEEEkeywords}

\section{Introduction}
 
The biological processes of cell lineage family tree that deployed in spatial and temporal are essential in the biomedical field including research, medicine and diagnosis a variety of diseases. Thus, quantitative analysis of cell behavior constructed on lineage tree tagged with appropriate measurements for the individual cell is the imperative basis for understanding the dynamics and structure of organisms \cite{r2.6}. Comprehensive and detailed information considering cell locality position, pathway, nucleus, formation, dissection, replicative and chronological lifespan can potentially be concluded from images using a combination of manual and automatic visualization tools \cite{r2.7}. Having an automatic tool that collects microscopic images and generates interactive visualization of the cell family tree with a quantitative comparison of cells would be a great requisite for many applications.\\

In the recent decade, crucial breakthroughs have been created and developed in the microscopy imaging of live-cell mechanisms such as fluorescent protein labeling, selective plane illumination microscopy (SPIM) and multiphoton laser scanning microscopy (MLSM)\cite{r2.7,r2.8}. Concurrently, many image processing algorithms have been developed for cell segmentation (e.g, semantic and instance), tracking and quantitative analysis \cite{r2.10}.\\

Since many of the experimental results from microscopic images are based on numerical numbers that require expertise to verify the validation of the outcome, visualization can offer an efficient path to study data such as yeast cell biological data. However, it has not adequately been developed in time-lapse microscopic images with interactive visualization. In fact, there is a knowledgeable gap between some biology areas and suitable visualization tools, in particular, visualizing microfluidics images interpreting the dynamic development of cells and lifespan. 

Microfluidics device is an ultra-small structure with microfluid channels offer faster and reliable results in comparison to traditional methods \cite {r2.1,r2.2}. Duo to their functionality, microscale, and nanoscale dimension, they can be used in many applications such as drug delivery, cell monitoring, cell division, virus inspection and so on \cite{r2.3,r2.4}. Time-lapse microfluidics images amplify biologists' ability to experimentally image live-cell during the developing progress \cite{r3}. Recently, a new network model of cellular aging was proposed based on the cellular aging process of yeast cells \cite{ref4.2}. For instance, Jo et al. \cite{\r13} investigated the high-throughput analysis of dividing yeast cells using microfluidic devices.\\ 

Cell division enumeration is a budding yeast process considering one of the essential factors in the aging process where single mother cells can be accomplished before inactivation, also known as replicative lifespan (RLS). Determining the lifespan of cells is a time-consuming process and in many cases is based on manual operation, such as cell micro-dissection \cite{ref08}. Accordingly, visualization of cell growth and division potentially can offer a great point of view to estimate RLS of cells (e.g, yeast cells) which would have an effective impact on many applications. Hence, structural information in high-resolution is one of the essential factors in understanding the molecular dynamics and cell functionalities such as microfluidics-based microscopy images. Yet, it is challenging to visualize and interpret the data gathered through microfluidics-based microscopy interactively.\\

To this end, we developed a circular plotting method, $\mu$Polar, to interpret cell movements and cellular division events at different time points from yeast cells microfluidics images. Our method is interactive and facile to utilize. We demonstrated the utility of our method to describe the events of dividing yeast cells over a time period. Our method could be potentially applied to other types of microfluidic devices including microscopic images.\\

The paper is organized as follows. In the results section, we concisely explain the data collection, feature extraction, and $\mu$Polar visualization analysis. In the discussion section, we focus on $\mu$Polar R package and $\mu$Polar functions. This section additionally covers some of $\mu$Polar plot examples including RLS. Finally, the conclusion section renders a summary of the proposed visualization method for microfluidics-based microscopic images.

 
%%%%%%%%%%%%%%%%%%%%%%%  to put note on pdf   %%%%%%%%%%%%%%%%%%%%%
%  One of the major \todo[inline]{change this to red}. However, ...
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\section{Results}

We developed an R package called "$\mu$Polar" for time-lapse microfluidics-based microscopic images. The sequence images could be visualized based on time, distance The cells area and RLS data could be included in the input dataset optionally. In addition, $\mu$Polar offers color-based cell tracking feature for visualization. The "$\mu$Polar" uses three libraries including "dplyr", "plyr" and "plotly". Fig.\ref{fig:table} shows trap reference point, cells area, cell distance and $\mu$Polar dataset formation. The "$\mu$Polar" function takes time-lapse "dataset", "time" (image time number), "distance" (reference-cell), "area" (cell area), "offset" (adjusting plot), "adjust" (adjusting cell size), "track" (active cell color code based tracking) and "RLS" (available RLS dataset. The package is set up for up to 12 color codes categorizing the number of available cells at each time-point and cell distance. Any number above 12 would be a random color. The color codes are adjustable and could be varied for different image types.  

\begin{figure*}
\centering
\includegraphics[width=\textwidth,height=10 cm]{Patterns/table.pdf}
\caption{ \textbf{ $\mu$Polar input dataset formation}. A sub-image labeled with distance and area at time-point 1 including a sub-image with reference point. A sample table of $\mu$Polar data format at time-point 1.}
\label{fig:table}
\end{figure*}


\subsection{Apply $\mu$Polar function for plotting}

Fig.\ref{fig:label} demonstrates an overview of $\mu$Polar function arguments for time-lapse images. The first argument is the time-lapse dataset that should be included time, distance, area (optional) and RLS (optional) which are required for an initial visualization as shown in Fig.\ref{fig:label}(a). The second and third arguments require the time and distance  column numbers from the dataset respectively. The fourth argument requires the cells area column number from the dataset which  can be set to "NA" if the area is unavailable. The default setting for the cell area is five. The fifth argument requires offset value adjusting the gap between the maximum cell radius of the plot and the plot out layer. Fig.\ref{fig:label}(c)) shows the comparison when "offset" set to 10 and when was initially set to "NA". This option is useful when there is an overcrowded region on the plot. The sixth argument relates to adjusting cell areas when the cells area dataset is available. Fig.\ref{fig:label}(d)) shows the comparison when the "adjust" set to 10 and  when was initially set to "NA". The sixth argument is for  cell color coding which can be applied as  "FALSE or "TRUE". This option is useful for cell tracking at each time-pints as illustrated in Fig.\ref{fig:label}(e).  The last argument requires RLS dataset column number and can be represented on the plot as "red star" as shown in (Fig.\ref{fig:label}(f)). The RLS argument could be set to "NA" in the RLS dataset is unavailable.


\begin{figure*}
\centering
\includegraphics[width=\textwidth,height=10 cm]{Patterns/option.pdf}
\caption{ \textbf{ $\mu$Polar function inputs for dataset, time, distance, area, offset, adjust, track and RLS}. (a) Initial $\mu$Polar plot without adjustment. (b) $\mu$Polar interactive feature. (c) The "offset" set to 10. (d) The area of cells is available and the "adjust" set to 10. (e) The "track" set to "TRUE". (f) the RLS of dataset is available and represented by "red star".}
\label{fig:label}
\end{figure*}


\subsection{Analysing $\mu$Polar plot}

 Fig.\ref{fig:read} is an overview of some prevalently events from time-lapse microfluidics images of yeast cells. Each of the sub-figures is a representation of most common events from cell visualization by $\mu$Polar. The black arrow in the event (a) pointing white color time-points is representing traps without any present cells which commonly occurs for traps close to the bottom of the microfluidics device due to shortening cell life-cycle. The event (b) portends the upcoming division event from potentially mother cell (mC) located inside the trap with a bud at trap outlet. This estimation is based on distance visualization between two cells (mc and bud). In the next event (c), the black dots are representing the single-cell separation cycle occurred after bud increased in size and the single-cell remains inside the trap. The event (d) shows overcrowded events when the number of cells in the image reaches more than three cells where each color are representing a different number of cells. The (e) and (f) events are good visualization examples of cell division which arguably notifies the occurrence of cell division. In the (e) event, the arrow on the Up region indicates that single-cell inside the trap (black dot) started division and there is not much cell development in size. However, the two arrows indicate that there is potentially a daughter cell (dC) attached to the cell inside the trap (mC) growing in size gradually until complete its division cycle, we called this event "dC below mC". The (f) event illustrates similar development in the opposite cells flow direction when a single call (mC) is located inside the trap in steady position and cell above it is growing gradually, we called this event "dC above mC". The (g) demonstrates an event that two cells have been close to each other for sometimes without much development in size. These cells considered as senescence time-points of cells. The (h) shows a single cell inside the trap has been in a steady position sometimes. This is most common at the end of the cell lifespan and considered as dead cell.


\begin{figure*}
\centering
\includegraphics[width=\textwidth,height=10 cm]{Patterns/read.pdf}
\caption{\textbf{ $\mu$Polar plot readability for some common cellular events}. Any event occurred below the reference line (passed trap outlet) denoted as "down" and any events occurred above the reference line (inside and above trap) denoted as "up". (a) The white dots (pointed with arrow) on the "red" reference line are indicating traps without cell. (b) The two arrows pointing two cells connecting with same line (sky blue color) are indicating that a single cell (black dot) started a division, cells are very close to each other (one inside the trap, one outside trap outlet). (c) The Black dots are representing a division after two cells completed the separation cycle. (d) Indication of an overcrowded cell event. (e) The arrow on the Up region indicates that there is a cell inside trap without much growing in size. That arrows on down region indicate that there is a single cell below and close to the cell inside the trap that gradually grew in size until complete separation. (f) The arrow on down region indicates that there is a single cell inside trap without much growing in size. The two arrows are indicating that there is a cell above and close to the cell inside the trap and growing in size until complete separation. (g) The arrows show two cells attached inside the trap and have not completed the separation cycle for sometimes. This is considered a cell senescence time-points. (h) The representation of a single cell that has been grown in steady size for sometimes. This is considered a dead cell time-points.}

\label{fig:read}
\end{figure*}


\subsection{$\mu$Polar for microfluidics images}

We have applied $\mu$Polar visualization tool to 104 traps time-lapse images. Fig.\ref{fig:compare} shows the visualization results for several traps including trap 2, trap 12, trap 22, trap 41, trap 50, trap 59, trap 73, trap 82, trap 90, trap 92. We chose these traps based on the investigation of cell development at the beginning, middle and end of the microfluidics device. In these images, each color represents several cells in a range of 1 to 6. The white, black, light blue, pink, green, purple, orange colors are representing 0 cells (blank trap),1 cell, 2 cells, 3 cells, 4 cells, 5 cells, and 6 cells respectively. For example, TP2 represents the time-lapse visualization of trap 2 where cell division occurred normally at the beginning between cells inside the trap and the trap outlet. The process gradually continued for first half of plot where there is a cell inside the trap and the growing cells at the trap outlet based on color representation at each time-point. The second half of the plot shows a clear overcrowded time-points represented with different color codes. The continuous black dots on TP12, TP22, TP73, TP82, and TP90, TP92 indicate that there is a single cell inside the trap, grown in size and has been occurring sometimes. These events could be considered as senescence or dead cell time-points when occurred at late time-points. 


\begin{figure*}
\centering
\includegraphics[width=\textwidth,height=10 cm]{Patterns/compare.pdf}
\caption{ \textbf{ $\mu$Polar visualization for several traps of microfluidics images}.}
\label{fig:compare}
\end{figure*}


Moreover, we evaluated some of $\mu$Polar plot from different trap numbers at different time-point with the corresponding image. For instance, S1\_Fig shows $\mu$Polar visualization of trap 8 at time-point 109. The plot indicates three cells (pink color)in the image that two cells are above reference point and single-cell is far below the reference point which considers as noise in this image. In another example, the S2\_Fig demonstrates trap 33 at time-point 11 when there are two cells available in the image. This is the early time-point of cell division when two cells (sky blue color) are very close to each other. On the other hand, the S3\_Fig demonstrates that there are two cells (sky blue color) reaching the end of the cell separation cycle according to the distance of the cells which corresponds to the number of cells in the image. The S4\_Fig illustrates a very interesting $\mu$Polar plot of trap 63 at time-point 340. The image shows that the cell is arguably at early development time-point, however, the $\mu$Polar plot indicates that the cell has been at this level for sometimes and considered as cell senescence time-point. In this research work, we tried to avoid the duplication of trap visualization, hence, most of the  given examples are from different trap number of microfluidics images.    


\subsection{$\mu$Polar for microscopic images}

In addition to the microfluidics-based microscopic images, we also utilized the $\mu$Polar to visualization the migrating mouse fibroblasts from images. We collected 37 time-lapse microscopic images (307x306) from \cite{ref05} as shown in Fig.\ref{fig:scopic}(a). The number of available cells in each image is in a range of 50 - 70 depending on the image time-point. For simplicity of the plotting, a section of these images cropped in the dimension of 121x121 based on average cell availability in a region of interest (Fig.\ref{fig:scopic}(b)). Correspondingly, feature extraction applied to these images collecting cells coordinate of each image in order of time. According to the observation, the cells are gradually migrating to the left direction of images over time. Thus, the reference point was chosen at the left-side edge of images (2,150) calculating cell distance. Fig.\ref{fig:scopic}(c) illustrates the $\mu$Polar plot where the cells number are in a range of 6 - 10 indicating by different color code. The "sky blue" color represents availability of 6 cells at the present time-point, the "light brown" color represents availability of 8 cells at the present time-point, the "light blue" color represents availability of 9 cells at the present time-point and the "purple" color represents availability of 10 cells at the present time-point. Fig.\ref{fig:scopic}(d) demonstrates cell size variation and cell tracking based on color labelling. For instance, the comparison between the closest cell to the center-point at time-point 1 (dark-red) and closest cell to center-point at time-point 37 (dark-red) shows that the cell migration from time-point 1 to time-point 37 is approximately 20 units. Since the number of images is 37,  $\mu$Polar color code indication shows a great visualization of cells corresponding to the movement and size variation. This is very useful for cell monitoring, counting division and cell tracking.       


\begin{figure*}
\centering
\includegraphics[width=\textwidth,height=10 cm]{Patterns/microscopic.pdf}
\caption{ \textbf{Migration of mouse fibroblasts cells from time-lapse microscopic images applying $\mu$Polar}.}
\label{fig:scopic}
\end{figure*}



\section{Discussion}

In this section, the simplicity and practicality of the $\mu$Polar tool in the contour expansion will be discussed. In Fig.\ref{fig:explain} example, cells are shown in five color codes depending on the number of available cells in the image. The black color represents a single cell, light blue represents two cells, pink color represents three cells, green color represents four cells and purple color represents five cells. At first glance, it appears that the number of cells in an image is two for the most of the time-points and many of the divisions are identifiable. For instance, the initial division started with dC below mC" event and changed to "dC above mC" event shortly (#1 and #2). During these time-points, there are small variations in the number of cells (pink and green colors) for a short period of time and followed by two cells. This process continued for some time-points and the number of cells increases from two (sky blue color) to three (pink color) cells for a longer period of time until there was not any cell available inside trap (#3). It is noticeable that the distance between cells and the reference point increased meaning cells are more likely located at top of the trap.

After this transition (2 cells to 3 cells), the division started again with "dC below mC" event and continued longer than a similar event at earlier time-pints. #4 represents an example of a completed division cycle point from two cells. These division cycle points have occurred several times in this example which this countability is useful for estimating the number of divisions by visualization. In addition, the overcrowded time-points (#6) can be identified by the corresponding color code. In general, dark colors represent cell location and lighter colors represent the distance between cells and reference points. This method is useful for tracking cells by the following cell with the same color at each time-point. For example, #5 shows a growing dC below mC which can be tracked down by dark-red color at each time point. Since the area of the cell can be applied to $\mu$Polar function, these cells growing time-points can also be visualized with corresponding cell size at each time-point as well as color codes. Senescence time-points (#7) are demonstrating that two grown cells have close to each other for sometimes without noticeable variation in cell size or division. The #8 represents a single cell inside the trap close to the end of the experiment work and potentially considered as a dead cell. This interpretation of cell development and comparability at each time-point can be beneficial for cell monitoring development and cell migration. 


\begin{figure*}
\centering
\includegraphics[width=\textwidth,height=10 cm]{Patterns/explain.pdf}
\caption{ \textbf{A trap example analysis with $\mu$Polar plot from time-points 1 - 391}.}
\label{fig:explain}
\end{figure*}

In another example from the different trap, Fig.\ref{fig:division} represents a situation that could lead to division miscounting during experimental work. This example shows a sequence of images from time-point 30 to 59 with a corresponding $\mu$Polar plot. Here, a single cell is only available at time-point 54 counting one division time-point. However, $\mu$Polar plot illustrates that the division also appears to occur between time-points 40-41 and time-point 47-48. The visualization of the development of the cells at trap outlet from time-points 40-41 and 47-48 can justify that the occurrence of the cell division is 4. This identification helps to have better division estimation especially when the interval between images is not short (e.g. 10 minutes). 

\begin{figure*}
\centering
\includegraphics[width=\textwidth,height=10 cm]{Patterns/division.pdf}
\caption{ \textbf{ Representation of microfluidics time-lapse images from time-points 30 to 59 with number of available cell(s)}. The time-point 35 (pink) indicates the number of cells decreasing from three to two. Time-points 40 to 41 and 47-48 indicate the occurrence of cell division. The time-point 53 represents a single cell inside trap (black).}
\label{fig:division}
\end{figure*}



\subsection{$\mu$Polar RLS}

$\mu$Polar has an additional option for replicative lifespan (RLS) visualization. This feature could be added to the plot if RLS dataset is available from the dataset. The RLS division point appears inside the plot out layer with a "star" sign  in "red" color. This feature gives a better understanding of division point with corresponding "red star".  Fig.\ref{fig:rls} demonstrates the RLS comparison between given RLS dataset and visually  counting RLS without prior information. We previously collected the experimental RLS data for each trap from \cite{ref02.2} and applied this information to the $\mu$polar. The black arrows inside the plot represent the division points from given RLS dataset based on experimental result which is 21 at corresponding time-points. Based on Fig.\ref{fig:read} $\mu$Polar guidelines referring to Fig.\ref{fig:read}, we estimated 20 of divisions with some unclear time-points which is close to the experimental result. Although there are some overcrowded time-points and large cells at the second half of the plot in this example, the plot shows the region of interest for counting RLS and determination point for RLS estimation in such an example. In $\mu$Polar plot visualization, the cell overlapping can be avoided by utilizing the "offset and "adjust" properly.       

\begin{figure*}
\centering
\includegraphics[width=\textwidth,height=10 cm]{Patterns/rlsTp1.pdf}
\caption{ \textbf{Identify the number of RLS from $\mu$Polar}. The "black arrows" are indicating the potential number of yeast cell division time-points from given experimental results (Estimate counting: 21). The "red stars" are indicating the number division time-points that can be identified by $\mu$Polar plot (Estimate counting: 20).}
\label{fig:rls}
\end{figure*}


Overall, the $\mu$Polar is a simple and useful tool for visualization of cell migration, cell monitoring, cell development, and cell division estimation from time-lapse microscopic images. The $\mu$Polar visualization can improve cell replicative lifespan accountability for the aging process. The comparison between yeast cells time-lapse microfluidics images and mouse fibroblasts cells time-lapse microscopic images demonstrates that $\mu$Polar can be a great tool for visualization time-lapse images with a single plot including comprehensive details. For instance, in the microfluidic images, we realized when cells becoming older over time, interestingly, the cell separation cycle becoming longer and there is a relationship between cell size (area) and separation cycle time for "dc below mC" event and "dC above mC" even. Moreover, $\mu$Polar could be a useful tool for microfluidic device design and quality evaluation, in particular, a microfluidic device with traps where each location of trap and shape can be visualized independently. Similarly, visualizing and tracking mouse fibroblasts cells show that cell development over time-points and the relationship between cell migration and cell size. Potentially, $\mu$Polar could be a great Visualization tool to compare a cell separation cycle time at its early lifespan and cell separation cycle time at its late lifespan.

  

\section{Star*Methods}

\subsection*{Data collection}
The microfluidics sequence images are procured from \cite{ref13} recent experimental work. The high-throughput yeast aging analysis (HYAA) chips device is a composed Polydimethylsiloxane (PDMS) model that contains 16 individual channels combined into four modules as shown in Fig.\ref{fig:micro}(a).

\begin{figure*}
\centering
\includegraphics[width=\textwidth,height=10 cm]{Patterns/fluidics.pdf}
\caption{ \textbf{HYAA microfluidics chips}. (a) Four modules (16 channels) microfluidic device. (b) A time-point sample of microfludics image shown with guidelines 10 - 70. (c) An example of time-lapse ( t1-t391) images for guidelines 10 - 20. (d) A sample of time-lapse (t1 - t391) partitioned images (60x60).}
\label{fig:micro}
\end{figure*}

All time-lapse microfluidics images are captured with IX-81 Olympus inverted microscope equipped with an Olympus camera (DP72 CCD) operating by cellSens software. The experimental process took around 96 hours at a temperature of 86$^{\circ}$F (30$^{\circ}$C). Each channel has 520 traps inside the microfluidic chamber and each row has 7 or 6 traps consecutively. Channels are labeled with numbers right-side for capturing guidelines. The order of rows could be different depending on capturing adjustment. For instant, if the camera positioned on guidelines 10 and 20, the first row of the captured image starts with 7 traps and so on. The experiment ended with capturing 391 images with 10 minutes interval. Images are captured at four locations of the microfluidic device ( Fig.\ref{fig:micro}(b)). 1st position: camera covered guidelines 10 - 20. 2nd position: camera covered guidelines 30 - 40. 3rd position: camera covered guidelines 50 - 60. 4th position: camera covered guideline 70. In this work, we have considered the dataset from 1st position (10 - 20) as shown in Fig.\ref{fig:micro}(c).

\subsection{Data Pre-processing}
Since microfluidics images have relatively low resolution and are considered a challenging task for visualization, we partitioned every single image to sub-images based on the individual trap. Traps are labeled from time-point 1 to time-point 391 as illustrated in Fig.\ref{fig:micro}. All sub-images partitioned concerning local traps boundary at 60x60 dimension. We used You Only Look Once (YOLOv3) \cite{ref20} machine learning model for image feature extraction. We collected 391 sequence sub-images for each trap, extract and label them as following: "time" (image time number), "distance (cell distance from reference point) and cell "area" (Fig.\ref{fig:table}). The cell features in each image are collected in no particular order (e.g, cell size, location). However, the features of each image are collected in order of time. For instance, Fig.\ref{fig:table} shows that the features of the first image input with four cells are collected indicating four rows for each cell and assigning time number (time) one for all of them. This labeling method continuously followed for the rest of the images.

\subsection{Distance calculation and time-points conversion}
The process of feature extraction carried on for 104 available traps as illustrated in Fig.\ref{fig:micro}(c). We then surcharged individual cell distance feature from the existing dataset by using the euclidean equation:

\begin{equation}
\begin{split}
d_i = \sqrt{(x_i -x_r)^2 + (y_i -y_r)^2}\\
\\
i =  [1,2,3,4,..., m ] \\
\\
\end{split}
\end{equation}

where $ d_i $ is the distance between the centroid point of cell and reference point. $ x_i $ and $ y_i $ are cells coordinate at each image. $ x_r $ and $ y_r $ are trap reference point. Generally, these points could be chosen at any point of image based on region of interest. We decided to choose the reference point based on cell flow direction and potential cell division point. $ i $ is image number corresponding to time-point and $ m $ is the maximum time-point equivalents to total number images. Here, $ x_r $ and  $ y_r $ are set to be 30 and 34 respectively at trap outlet as shown in $\nameref{S1_Fig}$. In a further step, we converted time to degree by using: 
 
\begin{equation}
\begin{split}
 degree = \sum_{i=1}^{n}{theta_i}\\
 \\
\end{split}
\end{equation}

here, $theta $ = \text{maximum degree}$/$\text{maximum time} and $ degree $ is accumulative degree of each radius and $ n $ is equal to maximum degree (e.g. 360). All distance values are considered as radius for cell movement interpretation.     


\subsection{$\mu$Polar input}

Fig.\ref{fig:table} illustrates an example of traps at time-points 1 and 2. Time-point 1 shows the label for distance and area of each cell. dL1, dl2, dL3, and dL4 are a distance from reference point. a1, a2, a3 and a4 are area of each cell. The table in Fig.\ref{fig:table} represents an example of the $\mu$Polar dataset format at two time-points. The dataset feature is labeled as following; "time" (image time number), "distance" (cell distance to reference point) and "area" (cell area)  which is optional and can be used for cell size variation visualization. Image at time-point 1 belongs to rows 1 - 4 of table dedicating image with four cells and image at time-point 2 belongs to row 5 - 7 dedicating image with three cells.



\subsection{$\mu$Polar visualization paradigm}

Fig.\ref{fig:polar}(a) is an interpretation of a trap sequence images from time-point 1 to time-point 391. The color representation is based on the number of available cell(s) in an image. Here, we used default setting where white color represents trap without cell, black color represents trap with one cell, green color represents trap with two cells, light-brown represents trap with 3 cells, blue color represents trap with 4 cells, orange color represents trap with 5 cells and sky-blue color represents trap with more than 5 cells. Furthermore, the red circle color represents trap reference line. For instance, time-point 275 to time-point 313 and time-point 353 to time-point 360 are represented in white color and laid on reference line (red). $ t  $ and  $ Ce $  are image time-point number and number of available cell(s) respectively. An example of $\mu$Polar plot given at time-point 21 where cell color shown in black since there only single-cell available. 

Fig.\ref{fig:polar}(b) shows a schematic of $\mu$Polar with some sample time-points from a trap. Images at time-points 1 - 6 are represented in $\mu$Polar plot. The theta for time-points 1 - 6 are 0 ,0.92, 1.84, 2.76, 3.68, 4.6 respectively. The plot has two regions separated with red circle, denoted as $ A $ region (cell above reference line) and $ B $ region (cell below reference). In this example, "White" color represents traps without cell located on reference line based on image theta, "black" color represents traps with single-cell located on radius line based on cell distance and image theta, "green" color represents traps with two cells located on radius line based on cell distance and image theta and "orange" color represents traps with three cells located on radius line based on cell distance and image theta. The cell size variation is based on the cell area. The number of cell(s) on each time-point radius indicates the number of cell (s) on each image. For instance, there is no cell at time-point 1, there are two cells at time-point 2, three cells at time-point 3, three cells at time-point 4, two cells at time-point 5 and one cell at time-point 6 respectively


\begin{figure*}
\centering
\includegraphics[width=\textwidth,height=10 cm]{Patterns/polar.pdf}
\caption{ \textbf{ A schematic and plot examples of $\mu$Polar}. (a) $\mu$Polar plot representation of microfludics time-lapse images from time-points 1 - 391. (b) $\mu$Polar schematic representation for time-lapse images from time-points 1 - 6 with indication of number of cell(s) with color codes.}
\label{fig:polar}
\end{figure*}


\section{Conclusion}
Microfluidics-based microscopy is becoming an increasingly popular research tool to monitor cellular events in biomedical research. One such application is the high-throughput analysis of dividing yeast cells. It is challenging to visualize and interpret the data gathered through microfluidics-based microscopy. Here, we developed a circular plotting method, $\mu$Polar, to visualize cell movements and cellular division events at hundreds of time points. Our method is interactive and easy to use. We demonstrated the utility of our method to describe the events of dividing yeast cells. Our method could be potentially applied to other types of microfluidic devices. This software is implemented in an R package $\mu$Po available through GitHub from https://github.com/merang/uPolar.



\section{Supplemental Information}


\paragraph*{S1 Fig.}
\label{S1_Fig}
{\bf  $\mu$Polar visualization of trap8 with corresponding microfluidics image}. 


% \begin{figure*}
% \centering
% \includegraphics[width=\textwidth,height=10 cm]{Patterns/bc8tp8.pdf}
% \caption{ \textbf{$\mu$Polar visualization for Trap08 microfluidics images}.}
% \label{S1_Fig}
% \end{figure*}


\paragraph*{S2 Fig.}
\label{S2_Fig}
{\bf  visualization of trap33 with corresponding microfluidics image}. 

% \begin{figure*}
% \centering
% \includegraphics[width=\textwidth,height=10 cm]{Patterns/bc8tp33.pdf}
% \caption{  \textbf{$\mu$Polar visualization for Trap33 microfluidics images}.}
% \label{S2_Fig}
% \end{figure*}



\paragraph*{S3 Fig.}
\label{S3_Fig}
{\bf  visualization of trap71 with corresponding microfluidics image}. 

% \begin{figure*}
% \centering
% \includegraphics[width=\textwidth,height=10 cm]{Patterns/bc8tp71.pdf}
% \caption{  \textbf{$\mu$Polar visualization for Trap71 microfluidics images}.}
% \label{S3_Fig}
% \end{figure*}




\paragraph*{S4 Fig.}
\label{S4_Fig}
{\bf visualization of trap71 with corresponding microfluidics image}. 

% \begin{figure*}
% \centering
% \includegraphics[width=\textwidth,height=10 cm]{Patterns/bc8tp63.pdf}
% \caption{ \textbf{$\mu$Polar visualization for Trap63 microfluidics images}.}
% \label{S4_Fig}
% \end{figure*}






% \paragraph*{S1 Table.}

% \label{S1_Table}
% {\bf ............} 


\subsection*{Acknowledgment}

The work is partially supported by the NSF CAREER award \#1453078 (transferred to \#1720215), NSF award \#1761839, a  start-up fund, internal CEACSE awards from the University of Tennessee at Chattanooga, and the computing facility of the SimCenter at the University of Tennessee at Chattanooga. We also acknowledge the support of NIH grants \#R01AG052507 and \#R42AG058368.


\subsection*{Author contributions}



\subsection*{Declaration on interests}

The authors declare no competing interests.  

\begin{thebibliography}{00}

\bibitem{r1}

CARPENTERA. E.: Image-based chemical screening.NatureChemical Biology 3, 8 (2007), 461–465


\bibitem{r2}
WOLLMANR., STUURMANN.: High throughput microscopy:From raw images to discoveries.Journal of Cell Science 120,21(2007), 3715–3722

\bibitem{r2.6}
Oates, A. C., Gorfinkiel, N., Gonzalez-Gaitan, M. & Heisenberg, C.-P. Quantitative approaches in developmental biology. Nat. Rev. Genet. 10, 517–530 (2009).


\bibitem{r2.7}
Amat, F. et al. Fast, accurate reconstruction of cell lineages from large-scale fluorescence microscopy data. Nat. Methods 11, 951–958 (2014).

\bibitem{r2.8}
Mavrakis, M., Rikhy, R., Lilly, M. & Lippincott-Schwartz, J. Fluorescence imaging techniques for studying Drosophila embryo development. Curr. Protoc. Cell Biol. 39, 4.18.1–4.18.43 (2008).

\bibitem{r2.9}
Buckingham, M. E. & Meilhac, S. M. Tracing cells for tracking cell lineage and clonal behavior. Dev. Cell 21, 394–409 (2011).

\bibitem{r2.10}
Olivier, N. et al. Cell lineage reconstruction of early zebrafish embryos using label-free nonlinear microscopy. Science 329, 967–971 (2010).


\bibitem{r2.2}
Farokhi H, Ghayesh MH (2018) Nonlinear mechanics of electrically actuated microplates. International Journal of Engineering Science 123: 197-213.

\bibitem{r2.1}
Ghayesh MH (2018) Mechanics of tapered AFG shear-deformable microbeams. Microsystem Technologies 24(4): 1743-1754.


\bibitem{r2.3}
Şimşek M (2016) Nonlinear free vibration of a functionally graded nanobeam using nonlocal strain gradient theory and a novel Hamiltonian approach. International Journal of Engineering Science 105: 12-27.

\bibitem{r2.4}
 Bhagat AAS, Bow H, Hou HW, Tan SJ, Han J, et al. (2010) Microfluidics for cell separation. Med Biol Eng Compu 48: 999-1014.

\bibitem{r2.5}
Shafiee H, Jahangir M, Inci F, Wang S, Willenbrecht RB, et al. (2013) Acute on-chip HIV detection through label-free electrical sensing of viral nano- lysate. Small 9(15): 2553-2563.


\bibitem{r3}
JENSENE. C.: Overview of live-cell imaging: Require-ments and methods used.The Anatomical Record 296, 1 (2013),1–8

\bibitem{ref01}

V. Longo, G. Shadel, M. Kaeberlein and B.  Kennedy, Replicative and Chronological Aging in Saccharomyces cerevisiae, Cell Metabolism, 2002,16(1):18-31.


\bibitem{ref02}
R. Eilsand and C. Athale, Computational imaging in cell biology, Cell Biol.2003, 16: 477-48.

\bibitem{ref02.2}
M. Ghafari,  et al. , Prototyping a family tree algorithm to estimate yeast replicative lifespan from time-lapse microfluidic images, IEEESouthEastConf. 2020, (awaiting for publication)


\bibitem{ref03}
D. Webb and A. Horwitz, New dimensions in cell migration, Cell Biol, 2003,  5: 690-692.

% \bibitem{ref04}
% M. Kaeberlein, K. Kirkland, S. Fields and B. Kennedy, Sir2-Independent Life Span Extension by Calorie Restriction in Yeast, PLoS Biology,,2004, 2(9):e296.

\bibitem{ref4.2}
H. Qin, Estimating network changes from lifespan measurements using a parsimonious gene network model of cellular aging, BMC Bioinformatics 20, 599 ,2019, doi:10.1186/s12859-019-3177-7









\bibitem{ref05}
H. Tsai et .al, Usiigaci: Instance-aware cell tracking in stain-free phase contrast microscopy enabled by machine learnin. 2019, doi.org/10.1016/j.softx.2019.02.007.

\bibi



tem{ref07}

D. Botstein and G. Fink, Yeast: An experimental organism for 21st Century biology. Genetics, 2001, 189(3): 695-704.


\bibitem{ref08}

K. Steffen, B. Kennedy and M Kaeberlein, Measuring replicative life span in the budding yeast, JoVE, 2009,  28: 1-5.

\bibitem{ref09}

M. Kaeberlein and B. Kennedy, Large-scale identification in yeast of conserved ageing genes. Mech Ageing Dev,2005, 126(1):17-21.

\bibitem{ref10}

D. Sinclair, K. Mills, and L. Guarente,  Aging in Saccharomyces cerevisiae, Annu. Rev,  Microbiol, 1998, 52: 533-560.


\bibitem{ref11}

S. Enfors  et al. , Physiological responses to mixing in large scale bioreactors, J Biotechnol,2001, 85: 175-185


\bibitem{ref12}

K. Chen, M. Crane and  M. Kaeberlein, microfluidics Technologies for Yeast Replicative Lifespan Studies. Mechanisms of ageing and development, 2017, 161(Pt B):262-269.


\bibitem{ref13}

 M. Jo, W. Liu, L. Gu, W. Dang and  L. Qin, High-throughput analysis of yeast replicative aging using a microfluidics system, Proc. Nati. Acad. Sci. USA, 2015, 112: 9364-9369.


\bibitem{ref14}

M. McCormick,  J.Delaney, M. Tsuchiya, et alA comprehensive analysis of replicative lifespan in 4,698 single-gene deletion strains uncovers conserved mechanisms of aging, Cell metabolism, 2015, 22(5):895-906.


\bibitem{ref15}
S. Pang,  et al. ,A novel YOLOv3-arch model for identifying cholelithiasis and classifying gallstones on CT images, PLoS ONE, 2019, 14(6): e0217647. https://doi.org/10.1371/journal. pone.021764

\bibitem{ref16}
C.  Laura, P. Hofmann, K. Drechselr  and S. Wesarg, Automatic detection of the nasal cavities and paranasal sinuses using deep neural networks, In IEEE 16th International Symposium on Biomedical Imaging,2019, pages 1154–1157. IEEE.

\bibitem{ref17}
J. Martin, et al., The impact of 2D cine MR imaging parameters on automated tumor and organ localization for MR-guided real-time adaptive radiotherapy. Physics in Medicine and Biology, 2018, 63(23):235005.

\bibitem{ref18}
S. Ramachandran, J. George, S. Skaria, and  V. Varun,  Using YOLO based deep learning network for real time detection and localization of lung nodules from low dose CT scans, In Kensaku Mori and Nicholas Petrick, editors, Medical Imaging 2018: Computer-Aided Diagnosis, 2018, volume 10575, page 53. SPIE.

\bibitem{ref19}
J. Redmon and A. Farhadi. YOLOv3: An Incremental Improvement. Technical report, University of Washington, 2018.

\bibitem{ref20}
Z. Zhao, P. Zheng, S. Xu, and X Wu, Object Detection With Deep Learning: A Review. IEEE Transactions on Neural Networks and Learning Systems, 2019, pages 1–21.


\end{thebibliography}



\end{document}
